### DISCUSSION TITLE
Fl_Image::scale problems

### DISCUSSION THREAD

**Discussion by CyprinusCarpio** [fltk/fltk#1018] [Q&A] [answered]
I'm trying to understand how to use `Fl_Image::scale` to display a scaled image. When I was making my program, I choose to use OpenGL to display a scalable canvas as OpenGL was something more familiar to me. But some time ago I got a report from a user that the texture isn't displayed as it exceeds the maximum size allowed by the GPU driver (16384x16384 in his case). After a hackfix involving binding a rescaled texture, I'm looking into having the texture drawn by the system api's (`StretchBlt` on win32 propably) instead. I've made a test program to see how it would work. It loads a png image and displays it scaled to the widget bounds.
```
#include <FL/Fl_Double_Window.H>
#include <FL/Fl_Box.H>
#include <FL/Fl_PNG_Image.H>

class Test : public Fl_Box
{
    Fl_PNG_Image* img;
public:
    Test(int X, int Y, int W, int H, char* L) : Fl_Box(X, Y, W, H, L)
    {
        img = new Fl_PNG_Image("test.png");
    }

    ~Test()
    {
        delete img;
    }
    
    void resize(int X, int Y, int W, int H) override
    {
        Fl_Box::resize(X, Y, W, H);
        img->scale(X, Y, false, true);
        redraw();
    }

    void draw() override
    {
        //Fl_Image* toDraw = img->copy(w(), h());
        //toDraw->draw(0, 0);
        //delete toDraw;
        img->draw(0, 0);
    }
};

int main(int argc, char** argv)
{
    Fl_Double_Window* win = new Fl_Double_Window(300, 200, "test");
    Test* t = new Test(0, 0, 300, 200, "");

    win->end();
    win->resizable(t);
    win->show(argc, argv);
    return Fl::run();
}
```
As you can see, the first time the `scale` function gets called is after the user resizes the window. Until then the image is displayed unscaled, correctly, as expected. But after rescaling, it disappears completely (both on win32 and on X). Commented out is the expensive workaround, which I'd rather avoid as in "production" it may involve copying upwards of several hundreds of megabytes. 
Why doesn't it work? I think I'm just doing something wrong, so I'm posting this here as opposed to a bug report.

**Comment by MatthiasWM** [comment, marked as answer]
Should this `img->scale(X, Y, false, true);` be `img->scale(W, H, false, true);` instead?

**Reply by CyprinusCarpio** [reply]
Of course that's the problem. Can't believe I missed that.
Drunk programming at 2300...

**Comment by Albrecht-S** [comment]
I'm closing this discussion because it got a valid (accepted) answer.
